pkg/skaffold/errors/err_map.go

```diff
@@ -87,3 +87,15 @@ var knownBuildProblems = []problem{
 		},
 	},
 }
+
+// Deploy errors in deployment phase
+var knownDeployProblems = []problem{
+	{
+		regexp:  re(fmt.Sprintf(".*%s.* Uanable to connect: .*", ClusterConnectErrPrefix)),
+		errCode: proto.StatusCode_DEPLOY_CLUSTER_CONNECTION_ERR,
+		description: func(error) string {
+			return "Deploy Failed."
```

Comment:
I think this gets turned into the log entry, which is then reflected to the output on the IDEs.  Maybe "Unable to connect to cluster"?

Reply:
The `description` is not used in the Event API but only for the console. 
The `ActionableErr.Message` which is the raw error is used. 

I made a conscious decision to send the raw error to IDEs to they could do any extra processing of the error message.
Maybe it makes sense to send the `description` or  we could change or add another field to the ` ActionableErr` e.g.
`ProcessedMessage` or `ExtractedMessage` along with the raw error text in `Message`.

Reply:
I only wondered if Skaffold would swallow the `Unable to connect` and instead report `Deploy Failed.`

Reply:
ah. got it. something like this
```
Deploy Failed. Check your connection for the cluster test_cluster.
```
Vs what we have right now. 

```

Deploy Failed. Could not connect to cluster test_cluster due to \"https://192.168.64.3:8443/version?timeout=32s\": net/http: TLS handshake timeout. 

Check your connection for the cluster.
```

Reply:
This change makes the message more readable / actionable.